### PR TITLE
Address derivation from private key

### PR DESCRIPTION
**Pull request by alejandro-isaza**

### CODE REVIEW

File: src/Coin.cpp [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        return Bitcoin::Bech32Address(HRP_LITECOIN, 0, data).string();
+
+    case TWCoinTypeNimiq:
+        return Nimiq::Address(data).string();
+
+    case TWCoinTypeRipple:
+        return Ripple::Address(data).string();
+
+    case TWCoinTypeDash:
+    case TWCoinTypeTron:
```
Comment (hewigovens) [inline src/Coin.cpp]: No TronAddress? I remember Tron uses sha3 instead of sha256 for keyhash

File: src/Coin.cpp [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        return Ripple::Address::isValid(string);
+
+    case TWCoinTypeDash:
+        return Bitcoin::Address::isValid(string, {TWP2PKHPrefixDash, TWP2SHPrefixDash});
+
+    case TWCoinTypeTezos:
+        return Tezos::Address::isValid(string);
+
+    case TWCoinTypeTron:
+        return Bitcoin::Address::isValid(string, {0x41});
```
Comment (hewigovens) [inline src/Coin.cpp]: Use a const variable for `0x41`?

File: src/Coin.cpp [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return DerivationPath(purpose(coin), coin, 0, 0, 0);
```
Comment (hewigovens) [inline src/Coin.cpp]: This is default case, I think we may need another methods:
1. accept `account`, `change`, `index`
2. only 3 levels for Ed25519

File: src/Coin.cpp [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        return Bitcoin::Bech32Address(privateKey.getPublicKey(PublicKeyType::secp256k1), 0, HRP_BITCOIN).string();
+
+    case TWCoinTypeBitcoinCash:
+        return Bitcoin::CashAddress(privateKey.getPublicKey(PublicKeyType::secp256k1)).string();
+
+    case TWCoinTypeDash:
+        return Bitcoin::Address(privateKey.getPublicKey(PublicKeyType::secp256k1), TWP2PKHPrefixDash).string();
+
+    case TWCoinTypeCallisto:
+    case TWCoinTypeEOS:
```
Comment (hewigovens) [inline src/Coin.cpp]: EOS should not Ethereum::Address

File: src/Coin.cpp [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        return Nimiq::Address(privateKey.getPublicKey(PublicKeyType::ed25519)).string();
+
+    case TWCoinTypeRipple:
+        return Ripple::Address(privateKey.getPublicKey(PublicKeyType::secp256k1)).string();
+
+    case TWCoinTypeTezos:
+        return Tezos::Address(privateKey.getPublicKey(PublicKeyType::secp256k1)).string();
+
+    case TWCoinTypeTron:
+        return Bitcoin::Address(privateKey.getPublicKey(PublicKeyType::secp256k1), 0x41).string();
```
Comment (hewigovens) [inline src/Coin.cpp]: Tron is base58 + sha3

File: src/Coin.cpp (hunk 1 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: We are missing a Tron::Address class...

File: src/Coin.cpp (hunk 2 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: Not sure what I should call this. It should be in Tron::Address once we have one.

File: src/Coin.cpp (hunk 3 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: We do support that, just none implemented yet.

File: src/Coin.cpp (hunk 4 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: Will remove this case since we don't support yet.

File: src/Coin.cpp (hunk 5 above)
Comment (alejandro-isaza) [reply src/Coin.cpp]: Again missing this class...

File: src/Coin.cpp (hunk 1 above)
Comment (hewigovens) [reply src/Coin.cpp]: Ok, There is a `TronAddress.swift`, that's why iOS still works

File: src/Coin.cpp (hunk 2 above)
Comment (hewigovens) [reply src/Coin.cpp]: Just tron address prefix?

**Pull request merged**